src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -634,130 +620,24 @@ QString QgsOgrProvider::ogrWkbGeometryTypeName( OGRwkbGeometryType type ) const
   return geom;
 }
 
-static OGRwkbGeometryType ogrWkbGeometryTypeFromName( const QString &typeName )
+OGRwkbGeometryType QgsOgrProviderUtils::wkbGeometryTypeFromName( const QString &typeName )
 {
-  if ( typeName == QLatin1String( "Point" ) ) return wkbPoint;
-  else if ( typeName == QLatin1String( "LineString" ) ) return wkbLineString;
-  else if ( typeName == QLatin1String( "Polygon" ) ) return wkbPolygon;
-  else if ( typeName == QLatin1String( "MultiPoint" ) ) return wkbMultiPoint;
-  else if ( typeName == QLatin1String( "MultiLineString" ) ) return wkbMultiLineString;
-  else if ( typeName == QLatin1String( "MultiPolygon" ) ) return wkbMultiPolygon;
-  else if ( typeName == QLatin1String( "GeometryCollection" ) ) return wkbGeometryCollection;
-  else if ( typeName == QLatin1String( "None" ) ) return wkbNone;
-  else if ( typeName == QLatin1String( "Point25D" ) ) return wkbPoint25D;
-  else if ( typeName == QLatin1String( "LineString25D" ) ) return wkbLineString25D;
-  else if ( typeName == QLatin1String( "Polygon25D" ) ) return wkbPolygon25D;
-  else if ( typeName == QLatin1String( "MultiPoint25D" ) ) return wkbMultiPoint25D;
-  else if ( typeName == QLatin1String( "MultiLineString25D" ) ) return wkbMultiLineString25D;
-  else if ( typeName == QLatin1String( "MultiPolygon25D" ) ) return wkbMultiPolygon25D;
-  else if ( typeName == QLatin1String( "GeometryCollection25D" ) ) return wkbGeometryCollection25D;
-  return wkbUnknown;
+  QgsWkbTypes::Type type = QgsWkbTypes::parseType( typeName );
+  return ( OGRwkbGeometryType ) type;
 }
 
 QStringList QgsOgrProvider::subLayers() const
 {
-  QgsDebugMsg( "Entered." );
   if ( !mValid )
   {
     return QStringList();
   }
-
   if ( !mSubLayerList.isEmpty() )
-    return mSubLayerList;
-
-  for ( unsigned int i = 0; i < layerCount() ; i++ )
   {
-    OGRLayerH layer = OGR_DS_GetLayer( ogrDataSource, i );
-    OGRFeatureDefnH fdef = OGR_L_GetLayerDefn( layer );
-    QString layerName = QString::fromUtf8( OGR_FD_GetName( fdef ) );
-    OGRwkbGeometryType layerGeomType = OGR_FD_GetGeomType( fdef );
-
-    // ignore this layer if a sublayer was requested and it is not this one
-    if ( mIsSubLayer &&
-         ( ( !mLayerName.isNull() && layerName != mLayerName ) ||
-           ( mLayerName.isNull() && mLayerIndex >= 0 && i != ( unsigned int )mLayerIndex ) ) )
-    {
-      QgsDebugMsg( QString( "subLayers() ignoring layer #%1 (%2)" ).arg( i ).arg( layerName ) );
-      continue;
-    }
-
-    if ( !mIsSubLayer && ( layerName == QLatin1String( "layer_styles" ) ||
```

Comment:
This was overlooked and has been build back in
* without **mIsSubLayer** since it is inside the static function